src/main/java/frc/robot/subsystems/ball/feeder/conveyor/ConveyorImpl.java

```diff
@@ -0,0 +1,101 @@
+package frc.robot.subsystems.ball.feeder.conveyor;
+
+import java.util.concurrent.TimeUnit;
+import ca.team3161.lib.robot.LifecycleEvent;
+import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
+import edu.wpi.first.wpilibj.Ultrasonic;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import frc.robot.RobotMap;
+import frc.robot.subsystems.ball.feeder.UltrasonicPoller;
+
+
+public class ConveyorImpl extends RepeatingPooledSubsystem implements Conveyor {
+
+    public enum ConveyorState {
+        FEEDING,
+        OFF,
+        UNLOADING,
+        PRIMING
+    }
+
+    WPI_TalonSRX conveyorController;
+    volatile ConveyorState state;
+
+    Ultrasonic topUltrasonic;
+    double topUltrasonicDistanceMM;
+
+    Ultrasonic bottomUltrasonic;
+    UltrasonicPoller bottomUltrasonicPoller;
+
+    ConveyorImpl() {
+        super(1, TimeUnit.SECONDS);
```

Comment:
This should probably be sped up, or else it will have up to a 1 second input latency between the driver/operator asking it to do something and it actually responding.

Reply:
I think basically every subsystem can just run at the default 50Hz, except for the Shooter which could benefit from being faster. If we actually start to see CPU load issues, which I doubt, then we can look at slowing some of them down.

Reply:
On that note, I/we should probably add a constructor for subsystems to default them to 50Hz.

Reply:
Yeah that would make sense, since that's what the looping methods in the Robot class run at.